From a0b3ce9ca5c69a945695c1c83dab7937a3d99b83 Mon Sep 17 00:00:00 2001
From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Sat, 5 May 2018 23:58:13 +0200
Subject: [PATCH] Dereference shared_ptr, fix for GCC8

Upstream-Status: Accepted [https://github.com/PointCloudLibrary/pcl/commit/a0b3ce9ca5c69a945695c1c83dab7937a3d99b83]

This patch has been generated with:
  `git format-patch -1 a0b3ce9ca5c69a945695c1c83dab7937a3d99b83`
in the pcl repository.

Signed-off-by: Lukas Bulwahn <lukas.bulwahn@gmail.com>
---
 segmentation/include/pcl/segmentation/ground_plane_comparator.h      | 2 +-
 segmentation/include/pcl/segmentation/plane_coefficient_comparator.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/segmentation/include/pcl/segmentation/ground_plane_comparator.h b/segmentation/include/pcl/segmentation/ground_plane_comparator.h
index e39354d..f96f38b 100644
--- a/segmentation/include/pcl/segmentation/ground_plane_comparator.h
+++ b/segmentation/include/pcl/segmentation/ground_plane_comparator.h
@@ -147,7 +147,7 @@ namespace pcl
       const std::vector<float>&
       getPlaneCoeffD () const
       {
-        return (plane_coeff_d_);
+        return (*plane_coeff_d_);
       }
 
       /** \brief Set the tolerance in radians for difference in normal direction between neighboring points, to be considered part of the same plane.
diff --git a/segmentation/include/pcl/segmentation/plane_coefficient_comparator.h b/segmentation/include/pcl/segmentation/plane_coefficient_comparator.h
index 9c94813..a21725a 100644
--- a/segmentation/include/pcl/segmentation/plane_coefficient_comparator.h
+++ b/segmentation/include/pcl/segmentation/plane_coefficient_comparator.h
@@ -141,7 +141,7 @@ namespace pcl
       const std::vector<float>&
       getPlaneCoeffD () const
       {
-        return (plane_coeff_d_);
+        return (*plane_coeff_d_);
       }
 
       /** \brief Set the tolerance in radians for difference in normal direction between neighboring points, to be considered part of the same plane.
-- 
2.7.4

